### PR TITLE
Fix update crosspost callback

### PR DESCRIPTION
**Pull request by jpaddison3**
Crosspost update callbacks were not updating the foreign server because they were exiting early. This was due to a mistaken use (I believe) of `document` as if it were the `oldDocument`. Instead the document is a merging of the `oldDocument` with the new `data` to create a "preview". The new data never differed from the "old document" (which was actually a preview of the new document), so the foreign server was never found to be in need of an update.

This PR:
 * Fixes the bug
 * Adds a comment about the deprecation of `document` (use `newDocument` instead)
 * Also deals (in a hacky way) with the gnarly state of re-drafting crossposts
   * This should prevent the issue which paged @jcmustin on Thursday
   * I'm okay with it being pretty hacky, as @oetherington is about to rewrite a bunch of this

Given that @oetherington is about to rewrite a bunch of this, perhaps the biggest deal in this PR is the calling attention to the bug and the marking of the `document` field as deprecated, and the rest of this was a ... suboptimal use of time. ¯\\\_(ツ)_/¯

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203861545124257) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by jpaddison3** [comment]
(Error is real.)

File: packages/lesswrong/server/fmCrosspost/crosspost.ts [hunk 1]
```diff
@@ -1,31 +1,43 @@
...
   }
 
   if (post.isEvent) {
+    logger ('post is an event, throwing')
     throw new Error("Events cannot be crossposted");
   }
 
   const user = await Users.findOne({_id: post.userId});
   if (!user || !user.fmCrosspostUserId) {
+    logger('user has not connected an account, throwing')
```
Comment (Will-Howard) [inline packages/lesswrong/server/fmCrosspost/crosspost.ts:34]: I don't really see the point of logging just before throwing an error

**Review by Will-Howard — APPROVED** [review]
Nice, I haven't tested this locally (because setting up crossposting is a hassle) but I'll trust that it works based on the bug you described.

I think having more debug logs is a step in the right direction... although I do feel that maybe this is going a bit too far, to the point of making it slightly hard to read the code in some places. I see the utility of permanent debug logs as being able to tell you _roughly_ where in the code things are happening, so that you can more easily add more logs or breakpoints to find the specific place

On a related note about logging (not relevant to this PR): I think it would be good to switch to using a proper logging library like [winston](https://github.com/winstonjs/winston) rather than `console.log`. There are a number of things this would help with:
 - proper debug/info/warning/error logging control
 - json formatted logs so that mutliline logs are handled properly in cloudwatch/datadog
 - associating logs with traces in datadog (you need json formatted logs for this)

File: packages/lesswrong/server/fmCrosspost/crosspost.ts (hunk 1 above)
Comment (jpaddison3) [reply packages/lesswrong/server/fmCrosspost/crosspost.ts:34]: Because sometimes errors are swallowed. ¯\\\_(ツ)_/¯

**Comment by jpaddison3** [comment]
This may have been too many logs.

**Comment by jpaddison3** [comment]
A claim I have is that you will often need to do debugging that passes through the vulcan guts. Keeping that part well-blanketed with debug logs is a help to the next time you have to do debugging that passes through the vulcan guts. Instead of adding and removing the same ones every time.

**Pull request merged**